### PR TITLE
Transfer ploting from tvb-multiscale/tvb-script to tvb-library - WIP

### PR DESCRIPTION
**Pull request by gabiflorea10**
This PR is subject of tasks HBP-27 and HBP-41

### CODE REVIEW

File: tvb/simulator/plot/utils/analyzers_utils.py
Comment (liadomide) [inline tvb/simulator/plot/utils/analyzers_utils.py]: This module should not be needed for plot Is it?

File: tvb/simulator/plot/utils/log_error_utils.py
Comment (liadomide) [inline tvb/simulator/plot/utils/log_error_utils.py]: This contradicts logging in tvb core modules. This I would not merge, and we should use the plotting with "get_logger" as it is in the entire tvb-library

File: tvb/simulator/plot/utils/data_structures_utils.py
Comment (liadomide) [inline tvb/simulator/plot/utils/data_structures_utils.py]: All modules under utils should be reviewed, and not copied here unless really used in plotting. I suspect many are not imported.

File: tvb/simulator/plot/timeseries_interactive.py
Comment (liadomide) [inline tvb/simulator/plot/timeseries_interactive.py]: I see 2 function from utils being used here. Maybe we can copy these alone from under utils, directly in the ts_interactive. Or create another single helper module with everything needed from utils.
The final decision much depends on the actual function needed

File: tvb/simulator/plot/head_plotter.py
Comment (liadomide) [inline tvb/simulator/plot/head_plotter.py]: This method would need at least model class Head under tvb-library. I do not see it. Is it here?
Alternatively, we could drop this method or change its signature, to receive aggregated DT from a Head as separate params

File: tvb/simulator/plot/config.py
Comment (liadomide) [inline tvb/simulator/plot/config.py:1]: This config module I would not like to see it copied entirely.
Could we deal only with FiguresConfig in a slightly modified form, thus we also know where to store figures (so part of the OutputConfig merge in the same class)?

File: tvb/simulator/plot/base_plotter.py
Comment (liadomide) [inline tvb/simulator/plot/base_plotter.py:451]: This looks odd. Should be re-written. Currently all is "tvb-plot" as we are under tvb-library

File: tvb/datatypes/time_series.py
Comment (liadomide) [inline tvb/datatypes/time_series.py:40]: Do we really do not have numpy already imported here?
I have not tried it, but it surprises me if that is the case. Maybe `import numy as np` ?

**Review by liadomide — CHANGES_REQUESTED** [review]
The transfer looks almost decent. See my comments for necessary changes.
I would prefer if we do not add more to the model (thus, do not create Head class in tvb-library, so either drop the plot for it, or change it) -- it refers to one of the comments in the code reviews.
In conclusion, I would like to see this one step further, as guided by the comments. Still we might decide not to merge, but it looks promising.

File: tvb/datatypes/time_series.py
Comment (gabiflorea10) [reply tvb/datatypes/time_series.py:40]: Surprisingly, numpy is not imported in this file. This is the reason for this import.

File: tvb/simulator/plot/utils/analyzers_utils.py
Comment (gabiflorea10) [reply tvb/simulator/plot/utils/analyzers_utils.py]: In this moment, it is used in time_series_plotter.py , class TimeSeriesPlotter, method _plot_ts_spectral_analysis_raster (file that is copied from tvb-multiscale). Probably, we have to investigate in depth if we need or not to have the module analyzers_utils.

File: tvb/simulator/plot/utils/log_error_utils.py
Comment (gabiflorea10) [reply tvb/simulator/plot/utils/log_error_utils.py]: This has been reviewed/rewritten and it can be viewed in a later commit.

File: tvb/simulator/plot/utils/analyzers_utils.py
Comment (liadomide) [reply tvb/simulator/plot/utils/analyzers_utils.py]: Ok. Then maybe make a smaller utils.py with only used method(s).
Depending on what will be left, we decide on a name for the new utils.
We could even consider putting the remaining function directly in module time_series_plotter.py (next to the place where it is used)

File: tvb/datatypes/time_series.py
Comment (liadomide) [reply tvb/datatypes/time_series.py:40]: Ok. Thanks for checking. Then it is perfectly ok to import it here.

File: tvb/simulator/plot/timeseries_interactive.py
Comment (gabiflorea10) [reply tvb/simulator/plot/timeseries_interactive.py]: These 2 functions has been copied directly in the timeseries_interacive.

File: tvb/simulator/plot/plotter.py
Comment (liadomide) [inline tvb/simulator/plot/plotter.py]: The same Head class is indirectly referred here also

File: tvb/simulator/plot/timeseries_interactive.py
Comment (liadomide) [inline tvb/simulator/plot/timeseries_interactive.py]: We should be py3 compatible only. 
Maybe we can drop this import?

**Review by liadomide — CHANGES_REQUESTED** [review]
It is looking better. Still some things to improve.
But now, I am almost convinced that this (after few other fixes) has potential to be merged.
We should hear what @maedoc and @dionperd have to say also.

File: tvb/simulator/plot/timeseries_interactive.py
Comment (gabiflorea10) [reply tvb/simulator/plot/timeseries_interactive.py]: Indeed, this can be dropped.

File: tvb/simulator/plot/base_plotter.py
Comment (gabiflorea10) [reply tvb/simulator/plot/base_plotter.py:451]: A possible solution can be seen in a comment bellow.

File: tvb/simulator/plot/utils/analyzers_utils.py
Comment (gabiflorea10) [reply tvb/simulator/plot/utils/analyzers_utils.py]: Ok. I created a utils.py with all the used methods from plot utils.

File: tvb/simulator/plot/utils/data_structures_utils.py
Comment (gabiflorea10) [reply tvb/simulator/plot/utils/data_structures_utils.py]: Ok. I created a utils.py with all the used methods from plot utils.

File: tvb/simulator/plot/head_plotter.py
Comment (gabiflorea10) [reply tvb/simulator/plot/head_plotter.py]: We have chosen to change the method signature to receive Head as separate parameters.

File: tvb/simulator/plot/plotter.py
Comment (gabiflorea10) [reply tvb/simulator/plot/plotter.py]: The proposed solution can be seen in a comment above.

File: tvb/simulator/plot/config.py
Comment (gabiflorea10) [reply tvb/simulator/plot/config.py:1]: I think, yes. A possible solution can be seen in a commit.

**Comment by liadomide** [comment]
I close this, as the code changes were moved on tvb-root repo

**Pull request closed without merging**
